contracts/Vault.vy

```diff
@@ -813,9 +825,13 @@ def deposit(_amount: uint256 = MAX_UINT256, recipient: address = msg.sender) ->
 @internal
 def _shareValue(shares: uint256) -> uint256:
     # Determines the current value of `shares`.
-    # NOTE: if sqrt(Vault.totalAssets()) > 1e37, this could potentially revert
-    # NOTE: using 1e3 for extra precision here, when decimals is low
-    return ((10 ** 3 * (shares * self._totalAssets())) / self.totalSupply) / 10 ** 3
+        # NOTE: if sqrt(Vault.totalAssets()) >>> 1e39, this could potentially revert
+    lockedFundsRatio: uint256 = (block.timestamp - self.lastReport) * self.lockedProfitDegration
```

Comment:
i wonder if on line 959 we can have two internal ShareValues, the complete one for pricePerShare, and if the user lastDeposit was within a certain time frame (add blocktimestamp on deposit), we don't need to add this calc and save the user some gas, this only applies to users that deposited briefly before the window of lastReport.

More of an optimization.